types/errors.go

```diff
@@ -64,23 +64,24 @@ var (
 const (
 	codeBadNumber terror.ErrCode = 1
 
-	codeDataTooLong           = terror.ErrCode(mysql.ErrDataTooLong)
-	codeIllegalValueForType   = terror.ErrCode(mysql.ErrIllegalValueForType)
-	codeTruncated             = terror.ErrCode(mysql.WarnDataTruncated)
-	codeOverflow              = terror.ErrCode(mysql.ErrDataOutOfRange)
-	codeDivByZero             = terror.ErrCode(mysql.ErrDivisionByZero)
-	codeTooBigDisplayWidth    = terror.ErrCode(mysql.ErrTooBigDisplaywidth)
-	codeTooBigFieldLength     = terror.ErrCode(mysql.ErrTooBigFieldlength)
-	codeTooBigSet             = terror.ErrCode(mysql.ErrTooBigSet)
-	codeTooBigScale           = terror.ErrCode(mysql.ErrTooBigScale)
-	codeTooBigPrecision       = terror.ErrCode(mysql.ErrTooBigPrecision)
-	codeWrongFieldSpec        = terror.ErrCode(mysql.ErrWrongFieldSpec)
-	codeTruncatedWrongValue   = terror.ErrCode(mysql.ErrTruncatedWrongValue)
-	codeUnknown               = terror.ErrCode(mysql.ErrUnknown)
-	codeInvalidDefault        = terror.ErrCode(mysql.ErrInvalidDefault)
-	codeMBiggerThanD          = terror.ErrCode(mysql.ErrMBiggerThanD)
-	codeDataOutOfRange        = terror.ErrCode(mysql.ErrWarnDataOutOfRange)
-	codeDuplicatedValueInType = terror.ErrCode(mysql.ErrDuplicatedValueInType)
+	codeDataTooLong              = terror.ErrCode(mysql.ErrDataTooLong)
+	codeIllegalValueForType      = terror.ErrCode(mysql.ErrIllegalValueForType)
+	codeTruncated                = terror.ErrCode(mysql.WarnDataTruncated)
+	codeOverflow                 = terror.ErrCode(mysql.ErrDataOutOfRange)
+	codeDivByZero                = terror.ErrCode(mysql.ErrDivisionByZero)
+	codeTooBigDisplayWidth       = terror.ErrCode(mysql.ErrTooBigDisplaywidth)
+	codeTooBigFieldLength        = terror.ErrCode(mysql.ErrTooBigFieldlength)
+	codeTooBigSet                = terror.ErrCode(mysql.ErrTooBigSet)
+	codeTooBigScale              = terror.ErrCode(mysql.ErrTooBigScale)
+	codeTooBigPrecision          = terror.ErrCode(mysql.ErrTooBigPrecision)
+	codeWrongFieldSpec           = terror.ErrCode(mysql.ErrWrongFieldSpec)
+	codeTruncatedWrongValue      = terror.ErrCode(mysql.ErrTruncatedWrongValue)
+	codeUnknown                  = terror.ErrCode(mysql.ErrUnknown)
+	codeInvalidDefault           = terror.ErrCode(mysql.ErrInvalidDefault)
+	codeMBiggerThanD             = terror.ErrCode(mysql.ErrMBiggerThanD)
+	codeDataOutOfRange           = terror.ErrCode(mysql.ErrWarnDataOutOfRange)
+	codeDuplicatedValueInType    = terror.ErrCode(mysql.ErrDuplicatedValueInType)
+	codeDatetimeFunctionOverflow = terror.ErrCode(mysql.ErrDatetimeFunctionOverflow)
```

Comment:
Where is it used?

Reply:
Line 113 use it as a key in map `typesMySQLErrCodes`.

Reply:
Line 113 use it as a key in map `typesMySQLErrCodes`.

Reply:
Then can we remove it in the map since there is no one use it to query the map?

Reply:
In `builtin_time.go`, line 2686 and line 2719 would return this specific error to client if it out of range.

Reply:
No, it is not. You can see the definition of `types.ErrDatetimeFunctionOverflow`, it is defined in terms of `mysql.XXX`.

Reply:
Oh, you are right, my mistake.
Now this error has been removed.